File: put-marbles-in-bags/put-marbles-in-bags.cpp
class Solution {
public:
    #define ll long long
    long long putMarbles(vector<int>& w, int k) {
        ll ans=0LL;
        int n=w.size();
        for(int i=0;i<n-1;i++) w[i]+=w[i+1];
        w.pop_back();
        sort(w.begin(),w.end());
        for(int i=0;i<k-1;i++) ans+= (ll)(w[n-i-2]-w[i]);
        return ans;
    }
};

File: put-marbles-in-bags/put-marbles-in-bags.py
class Solution(object):
    def putMarbles(self, w, k):
        ans=0
        n=len(w)
        for i in range(n-1):
            w[i]+=w[i+1]
        w.pop()
        w.sort()
        for i in range(k-1):
            ans += (w[n-2-i]-w[i])
        return ans